pkg/network/node/egressip.go

```diff
@@ -141,13 +158,70 @@ func getMarkForVNID(vnid, masqueradeBit uint32) string {
 	return fmt.Sprintf("0x%08x", vnid)
 }
 
+func (eip *egressIPWatcher) runIPAssignmentResync(stopCh <-chan struct{}) {
+	var addrChan chan netlink.AddrUpdate
+	addrSubscribe := func() error {
+		addrChan = make(chan netlink.AddrUpdate)
+		err := netlink.AddrSubscribeWithOptions(addrChan, stopCh, netlink.AddrSubscribeOptions{
+			ErrorCallback: func(err error) {
+				klog.Errorf("Failed during AddrSubscribe callback: %v", err)
+			},
+		})
+		// when AddrSubscribeWithOptions fails it does not close the addrChan
+		if err != nil {
+			close(addrChan)
+		}
+		return err
+	}
+
+	syncIPs := func() {
+		eip.tracker.Lock()
+		defer eip.tracker.Unlock()
+
+		eip.SyncIPAssignments()
+	}
+
+	subscribeErr := addrSubscribe()
+	if subscribeErr != nil {
+		klog.Error("Error during netlink subscribe: %v", subscribeErr)
+	}
```

Comment:
You need to return if there was an error, right? (If it failed on the first attempt, there's no good reason to expect it would succeed on future attempts...)

Reply:
I assumed we wanted to have `runIPAssignmentResync` run in the background the whole time SDN is up, if we return the resync will never happen.

I am not sure how probable the netlink subscribe failure is but is the first attempt any different from the subsequent ones?